netbox_netprod_importer/vendors/cisco/nxos.py

```diff
@@ -125,3 +120,43 @@ def get_detailed_lldp_neighbours(self):
                 "port": n["port_id"],
                 "chassis_id": n["chassis_id"]
             }
+
+    def get_detailed_cdp_neighbours(self):
+        """
+        Napalm does not support cdp
+
+        :return neighbours: [{
+                "local_port": local port name,
+                "hostname": neighbour hostname (if handled),
+                "port": neighbour port name,
+            }]
+        """
+        cmd = "show cdp neighbors detail" + self._driver_end_selection()
+
+        cmd_output = self.device.cli([cmd])[cmd]
+        neighbours = self._correct_and_convert_to_dict(cmd_output)["TABLE_cdp_neighbor_detail_info"][
+            "ROW_cdp_neighbor_detail_info"]
+
+        if isinstance(neighbours, dict):
+            neighbours = [neighbours]
+
+        for n in neighbours:
+            yield {
+                "local_port": n["intf_id"],
+                "hostname": re.sub("\(.*$", "", n["device_id"], count=1),
+                "port": n["port_id"],
+            }
+
+    def _driver_end_selection(self) -> str:
+        if type(self.device) is NXOSDriver:
+            self.device.device.api.cmd_method_raw = "cli"
+            return ""
+        else:
+            return " | json"
+
+    def _correct_and_convert_to_dict(self, cmd_output) -> dict:
+        if type(cmd_output) is not dict:
+            if not re.search("^{", cmd_output):
+                cmd_output = re.sub('^.+\. {', '{', cmd_output, count=1)
```

Comment:
Linked to what I said before, I will try to test on a real device tomorrow to understand correctly the changes.

Reply:
Here I have problems, at 6000 everything is fine, but 9000 problems arose.
The user has rights only to show *
# show lldp neighbors detail | json
SB: Permission denied. SB: Permission denied. SB: Permission denied. {"neigh_hdr":
If you do not use pipe, then there is no error.